### ISSUE TITLE
HASS.IO v0.97.1 Lovelace Card not displaying correct KNX Operation Modes

### ISSUE THREAD

**Issue by G8YTZ** [home-assistant/home-assistant-polymer#3517]
In the States UI thermostats appear with the ability to select one of the 4 KNX Operation Modes. With 0.97 the Lovelace Card only displays 3 Operation modes and these do not work to change the Operation mode. The picture shows the Graphic that is incorrect, but the "drill-down" is ok.
<img width="618" alt="Screenshot 2019-08-10 at 15 08 49" src="https://user-images.githubusercontent.com/35895759/62822837-e09b6880-bb80-11e9-8d3e-a5a5c94a3140.png">

**Comment by G8YTZ** [comment]
I did see that comment about reduction in functionality, but that cannot be true, frankly if this does not work I'm wasting my time going with Home Assistant.

KNX is the worldwide BMS standard with massive adoption, Google and the like are just toys.

**Comment by balloob** [comment]
No functionality was reduced, it's just accessible as a preset instead of an hvac mode.

**Comment by G8YTZ** [comment]
The whole Lovelace Thermostat interface is now basically useless, but the documentation does not say this. How is the current operation mode displayed? How is it usable from a finger on an iPhone or on a car dashboard or a touchscreen phone like the Gigaset Maxwell 10?
This was what was so brilliant - easy to use and not it does not work with the leading BMS system. That cannot be right.

**Comment by selfisch** [comment]
> In the Lovelace UI Thermostat card only 3 modes are shown, but on the STATES UI all 4 are shown and this work properly. Also the mode buttons no longer work on the Lovelace UI and the state colours are not shown either.
> 
> The Lovelace UI Thermostat card has been useless since v 0.96, yet it worked perfectly on previous versions.
> 
> The key issue is with the Lovelace Thermostat Card. The picture here shows only three operation modes available when there should be 4 like there always was before.
> 
> <img alt="Screenshot 2019-08-23 at 12 55 51" width="252" src="https://user-images.githubusercontent.com/35895759/63591028-83061380-c5a5-11e9-8cb4-c13363d6bbf7.png">

Now I got the point, sorry for that!
You are absolutely right.
I am currently not using that kind of card in lovelace ui, but hopefully those modes wont be taken away completely from the knx climate integration :-)

**Comment by G8YTZ** [comment]
Cool! here is what the thermostat page looks like (prior to 0.96) on my Tesla Screen/ Nabu Casa

![IMG_2041](https://user-images.githubusercontent.com/35895759/63622860-d8ffa900-c5ef-11e9-9506-1d5195e6d954.jpeg)

**Comment by G8YTZ** [comment]
A video also....

https://youtu.be/10nqj6f0uwc

**Comment by Mariusthvdb** [comment]
wow, looking mightily impressive. But please don't tell us you have 12 thermostats ..? (sorry if this diverges from the OP)

**Comment by G8YTZ** [comment]
The house is KNX, only diverges for things like Yale Conexis integration (via Smartthings Hub) external web sensors, HIK Vision Cameras, Cars etc. I took a KNX certification course 10 years ago when planning the house self-build.

The heating consists of a Mitsubishi Ecodan air-water Heat Pump, Theben HMT12 actuator that controls 11 underfloor zones and then there are 4 electric towels radiators in the bathrooms, so there are actually 15 zones, though several are dual heat zones.

The thing I like about KNX is that it's and open protocol and a "walled garden" - one configuration tool for all manufacturers and 100% compatibility between manufacturers is gauranteed. The design life of the kit is about 25 years, no software updates required. It just works and does not become obsolete.

Next integration challenge is the Resol solar thermal controller, might do the PV too, but right now I use a Zennio KEX KNX current sensor to measure power.

**Comment by G8YTZ** [comment]
Hi, is anyone looking at this issue since 0.96? It has completely broken the KNX thermostat integration. Someone has not thought these changes through, a major backward step that writes off HASS for use with KNX which is the largest professional BMS out there.

**Comment by iantrich** [comment]
The card shows modes, not presets. You can access the more-info dialog which shows the presets by clicking the top right icon. I don't consider this issue based on your description. The modes of the integration changed, not our display.

**Comment by G8YTZ** [comment]
Yes the Operation Modes don't work anymore. It has completely destroyed the UI with Lovelace how can the card show and control the operation modes correctly again? There must be a fix. Whatever had been done has ruined the interface.

**Comment by iantrich** [comment]
I see. People seemed to be focusing on the number of modes rather than the failure of the buttons.

So to be clear, all the mode buttons on the card no longer work for you but selecting from the drop-down down work?

**Comment by G8YTZ** [comment]
Yes, that's correct, also you see only 3 operation mode buttons and before there were 4. The picture on the Tesla Dash vs the other (post 0.96) pictures I took. The Lovelace card lends itself very well to a touchscreen, the drop-down does not.

**Comment by iantrich** [comment]
Let's not conflate the two issues. I think the buttons working is of importance. The modes was a bacoend choice which can be discussed in another issue.

**Comment by G8YTZ** [comment]
Nothing wrong with the back-end controls at all - the dropdown works as it always did. The issue as you say is that the Lovelace card no longer works correctly and does not display the correct operation modes. In my first picture I showed the dropdown was ok with all the modes.

**Comment by iantrich** [comment]
That drop-down is not modes, it's presets. You're conflating the two issues again. Presets will not be in the card.

**Comment by G8YTZ** [comment]
They are defined a "Operation Modes" in KNX. It worked properly before 0.96. Also how do you explain that the Card thermostat trace colours no longer change in the Lovelace Card? They are all grey.

**Comment by iantrich** [comment]
I'm not going to debate operation modes with you, that was a decision by the core backend devs to better align integrations. They are effectively presets/actions in terms of what the frontend sees. 

As for "traces", I don't know what that means and is again a different tangent.

**Comment by iantrich** [comment]
@pvizeli is what I'm saying accurate for climate 1.0?

**Comment by G8YTZ** [comment]
OK, look at the Tesla dashboard picture, you see green and orange dashboard traces according to KNX operation modes (call them presets if you like, but the correct KNX term is operation mode) Also look at why not all of the "presets" are displayed anymore there should be 4.
Whatever the backend devs have decided they have broken a very important integration and broken Lovelace too, so the best thing is to revert or add a switch in the software, but I thin k this is an oversight or bug.

**Comment by G8YTZ** [comment]
<img width="298" alt="Screenshot 2019-09-11 at 23 30 31" src="https://user-images.githubusercontent.com/35895759/64740114-2b93fd00-d4ec-11e9-82b8-9a3ee5fc7f37.png">

Now always grey and only 3 "presets"

**Comment by pvizeli** [comment]
I think the issue is:
Before 0.96 - Our Climate Object was based on the integration (in your case KNX)
Now - We have our own Climate Handling / The integration map this to

That means, whatever you KNX say, that does not mean it is mapped like this into Home Assistant.

That it shows wrong on UI is, because the KNX team on HA did not integrate our model correct. I move this issue back to Home Assistant and ping the KNX team. I can not promise that they are active and fix that soon.
